tests/e2e/framework/framework_test.go

```diff
@@ -134,7 +136,9 @@ func TestFailure(t *testing.T) {
 	c.Cleanup.RegisterCleanable(tc.s)
 	c.Cleanup.RegisterCleanable(tc.t)
 	tc.t.failRun = true
-	c.RunTest(tc.t)
+	if ret := c.RunTest(tc.t); ret == 0 {
+		t.Errorf("RunTest should have failed")
+	}
 	b := []int{1, 2, 4, 5}
```

Comment:
Could you also add some logs here saying like "Testing failure case, error expected..." so that it's less confusing for people?